src/AffectedUserFlow/CodeInput/CodeInputForm.tsx

```diff
@@ -49,10 +49,16 @@ const CodeInputForm: FunctionComponent = () => {
   const [isFocused, setIsFocused] = useState(false)
 
   const codeLength = 8
+  const codeIsInvalidLength = code.length !== codeLength
+  const codeContainsNonDigitChars = (code: string) => !code.match(/^\d+$/)
```

Comment:
Nice! I tend to prefer explicit returns but that's totally a matter of preference.
```
const codeContainsNonDigitChars = (code: string) => { return !code.match(/^\d+$/) }
```

Reply:
Just making sure re: @johnschoeman's comment, does `code.match(…)` not error if `code` is null?

Reply:
For the first use of codeContainsNonDigitChars in handleOnChangeText - we have an empty string check so that we don't show an error when the code is an empty string. This happens when deleting all the characters after typing.

For the second use as part of isDisabled, we do not need an empty string check because the valid length check handles that for us.

We should be good in both cases - since the code is never null, but an empty string.